LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/Regions/Region.cs

```diff
@@ -354,6 +354,11 @@ public DataRateIndex GetDRFromFreqAndChan(string datr)
             return DRtoConfiguration.FirstOrDefault(x => x.Value.DataRate.XpkDatr == datr).Key;
         }
 
+        public DataRateIndex GetDataRateIndex(DataRate datr)
+        {
+            return DRtoConfiguration.FirstOrDefault(x => x.Value.DataRate == datr).Key;
```

Comment:
What is the data rate cannot be found? We just return `DR0` (default for key) in that case. Do we want the following instead?

```c#
public DataRateIndex GetDataRateIndex(DataRate datr) =>
    DRtoConfiguration.FirstOrDefault(x => x.Value.DataRate == datr) is (var index, (not null, _)) ? index : throw new KeyNotFoundException("");
```